File: 0058-length-of-last-word/0058-length-of-last-word.py
class Solution:
    def lengthOfLastWord(self, s: str) -> int:
        i = len(s) - 1
        while i >= 0:
            while(i >= 0 and s[i] == " "):
                i -= 1
            if i < 0:
                break
            end = i
            while (i >= 0 and s[i] != " "):
                i -= 1
            start = i 
            return end - start

File: 0058-length-of-last-word/0058-length-of-last-word.java
class Solution {
    public int lengthOfLastWord(String s) {
        // StringBuilder ans = new StringBuilder();

        int i = s.length() - 1;

        while(i >= 0){
            while(i >= 0 && s.charAt(i) == ' '){
                i --;
            }
            if (i < 0){
                break;
            }
            int end = i;

            while (i >= 0 && s.charAt(i) != ' '){
                i --;
            }
            int start = i;

            return end - start;
        }
        return 0;
    }
}